### PR TITLE
feat: implement `-context-only`

### PR DESCRIPTION
**Pull request by mattdowdell**
As described in #13, third-party implementations of `slog.Handler` can choose to take values out of a context to supplement the attributes already used. An example of this in my day job is adding trace IDs to enhance serviceability.

This change adds the ability to optionally identify `slog` methods that do not take a context and so prevent the ability to take values from that context. Any matches can be resolved by appending `Context` to the method. For example, `slog.Info` becomes `slog.Context`.

### CODE REVIEW

**Comment by tmzane** [comment]
Looks great! I appreciate you added the documentation as well :)

File: sloglint.go [hunk 1]
```diff
@@ -113,6 +116,14 @@ func run(pass *analysis.Pass, opts *Options) {
 			return
 		}
 
+		if opts.ContextOnly {
+			fnName := fn.FullName()
+
+			if !strings.HasSuffix(fnName, ".Log") && !strings.HasSuffix(fnName, "Context") {
```
Comment (tmzane) [inline sloglint.go:122]: WDYT if we check whether the first element of the `call.Args` slice is a context instead? Then no `.Log` special handling would be needed 🤔

Comment (mattdowdell) [reply sloglint.go]: The thought did cross my mind as I was writing it originally.

I've wwitched to checking the type of first arg, and added a couple of variations to the tests to ensure identifiers (`ctx`) and call expressions (`context.Background()`, etc.) behave the same. I think the test cover all the bases, but let me know if you can think of any other edge cases to prove out.

File: sloglint.go [hunk 2]
```diff
@@ -113,6 +115,13 @@ func run(pass *analysis.Pass, opts *Options) {
 			return
 		}
 
+		if opts.ContextOnly {
+			arg, ok := pass.TypesInfo.Types[call.Args[0]]
+			if ok && arg.Type.String() != "context.Context" {
+				pass.Reportf(call.Pos(), "methods that do not take a context should not be used")
```
Comment (mattdowdell) [inline sloglint.go:121]: I was a little unsure of the error message here. I tried to match the existing format of "X should not be Y", but the double negative irked me. Something like "only use methods that take a context" might be easier to understand?

Comment (tmzane) [reply sloglint.go:121]: How about "methods without a context should not be used"?

File: sloglint.go (hunk 1 above)
Comment (tmzane) [reply sloglint.go]: Not sure we need to cover different context cases (`context.Background()`, `context.TODO()`, etc), since we only check the type, and it is always the same.

**Review by tmzane — APPROVED** [review]

**Comment by tmzane** [comment]
@mattdowdell I've updated the report message and rearranged the options slightly. Thank you for the contribution!
